lib/tests/gather-fragment-imports-test.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+const { expect } = require('chai');
+const path = require('path');
+const fs = require('fs');
+const gatherFragmentImportsTest = require('../gather-fragment-imports');
+const FixturifyProject = require('fixturify-project');
+
+const fragmentCapture = /(fragment (\w*).*\{[\w\n\s]*)\}/gm;
+
+function* fakeParser(src) {
+  let fragmentSnippet;
+
+  while ((fragmentSnippet = fragmentCapture.exec(src)) !== null) {
+    yield {
+      name: {
+        value: fragmentSnippet[2],
+      },
+      fragmentSnippet: fragmentSnippet[0],
+    };
+  }
+}
+
+describe('gather-fragment-imports', function () {
+  let basedir;
+
+  beforeEach(function () {
+    const project = new FixturifyProject('my-example', '0.0.1', project => {
+      project.files['apple.graphql'] = `
+fragment apple on User {
+  name
+}`;
+      project.files['orange.graphql'] = `
+fragment orange on User {
+  name
+}`;
+      project.files['parent.graphql'] = `
+#import './apple.graphql'
+
+fragment parent on User {
+  name
+}`;
+      project.files['double-import.graphql'] = `
+#import './apple.graphql'
+#import './orange.graphql'
+
+fragment parent on User {
+  name
+}`;
+      project.files['nested-1.graphql'] = `
+#import './nested-2.graphql'
+
+fragment nested1 on User {
+  name
+}`;
+      project.files['nested-2.graphql'] = `
+#import './nested-3.graphql'
+
+fragment nested2 on User {
+  name
+}`;
+      project.files['nested-3.graphql'] = `
+fragment nested3 on User {
+  name
+}`;
+      project.files['nested.graphql'] = `
+#import './nested-1.graphql'
+
+fragment parent on User {
+  name
+}`;
+      project.files['from-dependency.graphql'] = `
+#import 'my-dependency/_dependency-fragment.graphql'
+query {
+  myQuery {
+    ...FromDependency
+  }
+}`;
+
+      project.addDependency('my-dependency', '0.0.1', dependency => {
+        dependency.files['_dependency-fragment.graphql'] = `
+fragment FromDependency on User {
+  name
+}`;
+      });
+    });
+    project.writeSync();
+    basedir = project.baseDir;
+  });
+
+  it('handles single import', function () {
+    const result = gatherFragmentImportsTest({
+      source: fs.readFileSync(path.join(basedir, 'parent.graphql'), 'utf8'),
+      sourceLocation: path.join(basedir, 'parent.graphql'),
+      resolveImport: require('resolve').sync,
+      fragmentParserGenerator: fakeParser,
+      throwIfImportNotFound: false,
+    });
+
+    expect(Object.keys(result).length).to.eql(1);
```

Comment:
can we make more precise assertions? `deep.equal` should be fine here since we're providing a test parse generator so shouldn't have to worry about extraneous properties